### PR TITLE
Version 2

### PR DESCRIPTION
**Pull request by anbraten**
## TODO

- [x] deploy nats-server for preview & testing
- [x] search
- [x] cache (done by jetstream)
- [x] data ttl, remove from app store / cache if outdated (done by jetstream)
- [x] subscriptions list with ttl and ping (done by jetstream and `subscriptions` handler)
- [x] restore subscriptions after reconnect (done by nats)
- [x] retry connection after error
- [ ] retry subscription
- [x] handle subscribe before connect
- [x] stops
- [x] trips
- [x] service-worker
- [x] create nats streams on server start (with json config)
- [ ] add differentiated streams (short retention for vehicles/trips, longer for stops)
- [x] prettier trip display
- [x] favorites
- [ ] deployment for kvg collector
- [x] make frontend nats servers configurable
- [x] stop alerts
- [x] show message on empty stop page
- [x] fix click listener in mobile popup
- [ ] extract ais collector
- [x] map selected styling
- [x] style service worker popup
- [x] style button
- [x] ~~style popup actions (open map)~~
- [x] flyTo vehicle marker on vehicle location change
- [x] loading animation
- [x] fix buggy time display width in bus popup when time includes `11`
- [x] fix empty osm attribution on mobile
- [x] ~center search results on desktop~
- [x] news popup
- [x] z index search overlay
- [x] open search again after writing 
- [x] favorites as popup default openend
- [x] import old favorites
- [x] remove `offlineReady` dialog?

### CODE REVIEW

File: collectors/kvg/main.go [hunk 1]
```diff
@@ -67,9 +77,69 @@ func main() {
...
+		if err := json.Unmarshal([]byte(msg.Data), &consumerEvent); err != nil {
+			log.Fatalf("Parse response failed, reason: %v \n", err)
+		}
+		if subjects[consumers[consumerEvent.Consumer]] > 1 {
+			subjects[consumers[consumerEvent.Consumer]]--
+		} else {
+			delete(subjects, consumers[consumerEvent.Consumer])
+		}
+		delete(consumers, consumerEvent.Consumer)
+		subscriptions = []string{}
```
Comment (anbraten) [inline collectors/kvg/main.go:131]: @lukashass Please add a `sync.Mutex` for `subscriptions`, `subjects` and `consumers` and lock if you change data.

Comment (lukashass) [reply collectors/kvg/main.go:131]: @anbraten What are the reasons for locking here?

One good reason I see is concurrent access by a callback writing into `subscriptions` and our scheduled collectors reading from `subscriptions`.

Another case I'm unsure about is concurrent access by two callbacks. nats is already doing some locking inside the subscriptions: https://github.com/nats-io/nats.go/blob/84e8b4dda7cca442de11abab3615665e7e3e258a/nats.go#L2615

Comment (lukashass) [reply collectors/kvg/main.go]: Added sync.Mutex in 26ea85f

**Comment by lukashass** [comment]
@anbraten I think it might be a good idea to extract dark mode and AIS collector to do them later?

File: app/src/views/Search.vue [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    <AppBar v-model:search-input="searchInput" />
+
+    <div class="flex flex-col mt-16 mx-2 w-full overflow-y-auto">
+      <div v-if="searchResults.length === 0" class="m-auto max-w-52 text-center text-xl">
+        <p>Suche nach einer Haltestelle oder einem Fahrzeug</p>
+      </div>
+      <router-link
+        v-for="searchResult in searchResults"
+        :key="searchResult.refIndex"
+        :to="{ name: 'map-marker', params: { markerType: searchResult.item.type, markerId: searchResult.item.id } }"
```
Comment (anbraten) [inline app/src/views/Search.vue:12]: @lukashass What's your suggestion to map `searchResult.item.type` to `stop` or `vehicle` ...?

**Review by lukashass — APPROVED** [review]

**Pull request merged**